_episodes/07-vector-data-in-python.md

```diff
@@ -0,0 +1,351 @@
+> However, in the GIS world, it is custom to work with coordinate tuples where the first component is aligned with the east/west direction and the second component is aligned with the north/south direction.
+> Multiple software packages thus implement this convention also when dealing with EPSG:4326.
+> As a result, one can encounter vector files that implement either convention - keep this in mind and always check your datasets!
+{: .callout}
+## Modify the geometry of a GeoDataFrame
+
+Sometimes we need to modify the `geometry` of a `GeoDataFrame`. For example, as we have seen in the previous exercise **Investigate the waterway lines**, the latitude and longitude are flipped in the vector data `waterways_nl`. This error needs to be fixed before performing further analysis.
+
+Let's first take a look on what makes up the `geometry` column of `waterways_nl`:
+
+~~~
+waterways_nl['geometry']
+~~~
+{: .language-python}
+~~~
+0     LINESTRING (52.41810 4.84060, 52.42070 4.84090...
+1     LINESTRING (52.11910 4.67450, 52.11930 4.67340...
+2     LINESTRING (52.10090 4.25730, 52.10390 4.25530...
+3     LINESTRING (53.47250 6.84550, 53.47740 6.83840...
+4     LINESTRING (52.32270 5.14300, 52.32100 5.14640...
+                            ...
+86    LINESTRING (51.49270 5.39100, 51.48050 5.39160...
+87    LINESTRING (52.15900 5.38510, 52.16010 5.38340...
+88    LINESTRING (51.97340 4.12420, 51.97110 4.12220...
+89    LINESTRING (52.11910 4.67450, 52.11850 4.67430...
+90    LINESTRING (51.88940 4.61900, 51.89040 4.61350...
+Name: geometry, Length: 91, dtype: geometry
+~~~
+{: .output}
+
+Each row is a `LINESTRING` object. We can further zoom into one of the rows, for example, the third row:
+
+~~~
+print(waterways_nl['geometry'][2])
+print(type(waterways_nl['geometry'][2]))
+~~~
+{: .language-python}
+~~~
+LINESTRING (52.100900002 4.25730000099998, 52.1039 4.25529999999998, 52.111299999 4.24929999900002, 52.1274 4.23449999799999)
+<class 'shapely.geometry.linestring.LineString'>
+~~~
+{: .output}
+
+As we can see in the output, the `LINESTRING` object contains a list of coordinates of the vertices. In our situation, we would like to find a way to flip the x and y of every coordinates set. A good way to look for the solution is to use the [documentation](https://shapely.readthedocs.io/en/stable/manual.html) of the `shapely` package, since we are seeking to modify the `LINESTRING` object. Here we are going to use the [`shapely.ops.transform`](https://shapely.readthedocs.io/en/stable/manual.html?highlight=shapely.ops.transform#shapely.ops.transform) function, which applies a self-defined function to all coordinates of a geometry.
+
+~~~
+import shapely
+
+# Define a function flipping the x and y coordinate values
+def flip(x, y):
+    return y, x
+
+# Apply this function to all coordinates and all lines
+geom_corrected = [shapely.ops.transform(flip, line) for line in waterways_nl['geometry']]
```

Comment:
Good point @rbavery ! I have modified the implementation, using a lambda function to keep it concise (it might also be a good way to introduce these) - what do you think?